src/Diagnostics.DataProviders/DataProviders/Base/SupportObserverDataProviderBase.cs

```diff
@@ -63,16 +63,38 @@ public Task<dynamic> GetResource(string resourceUrl)
                 throw new FormatException(exceptionMessage, ex);
             }
 
-            if (uri.Host.Contains(allowedHosts[0]) || uri.Host.Contains(allowedHosts[1]) || Configuration.ObserverLocalHostEnabled)
+            if (uri.Host.Contains(allowedHosts[0]) || uri.Host.Contains(allowedHosts[1]))
             {
                 return GetWawsObserverResourceAsync(uri);
             }
+            else if (Configuration.ObserverLocalHostEnabled && TryGetLocalObserverRoute(uri, out var newUri))
+            {
+                return GetWawsObserverResourceAsync(newUri);
+            }
             else
             {
                 return GetSupportObserverResourceAsync(uri);
             }
         }
 
+        private static bool TryGetLocalObserverRoute(Uri uri, out Uri localObserverUri)
+        {
+            localObserverUri = uri;
+
+            if (!uri.AbsolutePath.StartsWith("/observer"))
+            {
+                return true;
+            }
+
+            if (uri.AbsolutePath.Contains("subscriptions") && (uri.AbsolutePath.EndsWith("webspaces") || uri.AbsolutePath.EndsWith("sites")))
```

Comment:
Why only a subset of support-bay-api URLs?

Reply:
`/api/subscriptions/{}/webspaces` & `api/subscriptions/{}/sites` work in selfhost Observer
`/api/subscriptions/{}/hostingEnvironments/{}/deployments` isn't found in selfhost Observer, 404

Reply:
Can we move the business logic of this API from support-bay-api into the DiagnosticController on SelfHostObserver. I believe I built this API long ago and it is a a few lines of code.